src/object/conveyor_belt.cpp

```diff
@@ -0,0 +1,163 @@
+//  SuperTux
+//  Copyright (C) 2022 Raoul1808 <raoulthegeek@gmail.com>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "object/conveyor_belt.hpp"
+
+#include "badguy/walking_badguy.hpp"
+#include "sprite/sprite.hpp"
+#include "sprite/sprite_manager.hpp"
+#include "supertux/direction.hpp"
+#include "math/util.hpp"
+#include "util/reader_mapping.hpp"
+
+ConveyorBelt::ConveyorBelt(const ReaderMapping &reader) :
+  MovingObject(reader), // TODO: sprite
+  ExposedObject<ConveyorBelt, scripting::ConveyorBelt>(this),
+  m_running(true),
+  m_direction(Direction::LEFT),
```

Comment:
Direction member variables are mostly named `m_dir` in the code, so this one could be renamed to `m_dir` as well. Do this only if you wish to.